lms/djangoapps/discussion_api/serializers.py

```diff
@@ -245,22 +259,30 @@ def get_comment_list_url(self, obj, endorsed=None):
         )
 
     def get_endorsed_comment_list_url(self, obj):
-        """Returns the URL to retrieve the thread's endorsed comments."""
+        """
+        Returns the URL to retrieve the thread's endorsed comments.
+        """
         return self.get_comment_list_url(obj, endorsed=True)
 
     def get_non_endorsed_comment_list_url(self, obj):
-        """Returns the URL to retrieve the thread's non-endorsed comments."""
+        """
+        Returns the URL to retrieve the thread's non-endorsed comments.
+        """
         return self.get_comment_list_url(obj, endorsed=False)
 
     def get_comment_count(self, obj):
-        """Increments comment count to include post and returns total count of
-        contributions (i.e. post + responses + comments) for the thread"""
+        """
+        Increments comment count to include post and returns total count of
+        contributions (i.e. post + responses + comments) for the thread
+        """
         return obj["comments_count"] + 1
 
     def get_unread_comment_count(self, obj):
-        """Increments comment count to include post if thread is unread and returns
-        total count of unread contributions (i.e. post + responses + comments) for the thread"""
-        if not obj["read"]:
+        """
+        Increments comment count to include post if thread is never read and returns
+        total count of unread contributions (i.e. post + responses + comments) for the thread.
+        """
+        if not obj["read"] and obj["comments_count"] == obj["unread_comments_count"]:
```

Comment:
Case 2 and Case 3 will use `unread_comments_count` returned from `cs_comments_service` as is.
For Case 1; we need to increment `unread_comments_count` returned from `cs_comments_service` with 1.